test/sql/append.sql

```diff
@@ -0,0 +1,74 @@
+\ir include/create_single_db.sql
+
+-- create a now() function for repeatable testing that always returns
+-- the same timestamp. It needs to be marked STABLE
+CREATE OR REPLACE FUNCTION now_s()
+RETURNS timestamp LANGUAGE PLPGSQL STABLE AS
+$BODY$
+BEGIN
+     RETURN '2017-08-22T10:00:00'::timestamp;
+END;
+$BODY$;
+
+CREATE OR REPLACE FUNCTION now_i()
+RETURNS timestamp LANGUAGE PLPGSQL IMMUTABLE AS
+$BODY$
+BEGIN
+     RETURN '2017-08-22T10:00:00'::timestamp;
+END;
+$BODY$;
+
```

Comment:
Actually, the function I am using to fold stable functions (`estimate_expression_value()`) is described as "Fold stable, as well as immutable, functions to constants". Thus, I think it is safe to apply this modification to restriction clauses that have volatile functions in it, because it might also have stable functions that we can fold. For instance, let's say the WHERE clause has a `time > now() - interval '1 day'` and a `id = random()`, then we'd like to optimize that, no?